migrations/README.rst

```diff
@@ -168,7 +168,24 @@ For doing that the following command might be used:
 
     alembic downgrade head REVISION_ID
 
-| If you would like to use relative identifiers, such as :code:`alembic downgrade head -1`, you might check it out: `Relative Identifiers <https://alembic.sqlalchemy.org/en/latest/tutorial.html#relative-migration-identifiers>`_
+| If you would like to use relative identifiers, such as :code:`alembic downgrade -1`, you might check it out: `Relative Identifiers <https://alembic.sqlalchemy.org/en/latest/tutorial.html#relative-migration-identifiers>`_
+|
+| **Note-4:** During the migration of SQLite Database, it's possible to see this error:
+
+.. code-block:: bash
+
+    AddGeometryColumn() error: "UNIQUE constraint failed: geometry_columns.f_table_name, geometry_columns.f_geometry_column"
+
+You can ignore this error because it says that there is already a geometry column entity for your table that has a Geometry column (i.e. States, Geometries, Contact). This error happens because SQLite doesn't support *ALTER TABLE* statement.
+Instead, it creates a new one, copies values from the previous table, and drops the existing table. However, during the creation of a new table, it tries to push the Geometry column to the :code:`geometry_columns` table again.
+
+| **Note-5:** If you change a column's name, you should implement different classes for :code:`upgrade()` and :code:`downgrade()` methods. Otherwise, this class breaks your upgrade because it doesn't have that column to rename.
+|
+| An example can be found in :code:`migrations/sqlite_versions/2020-06-03_f103f27c4575_change_pennant_to_identifier_and_make_.py`. What you should do is as follows:
+
+- :code:`copy_from` post-hook will create all necessary tables. Copy the table that has a column change to :code:`upgrade()` and :code:`downgrade()` methods.
+- Remove the global class.
```

Comment:
I'm rather confused about Note 5. I don't understand why we have to copy a definition of the table in there for this, and I'm not sure what the copy_from post-hook will do, or what it means to remove the global class (presumably not delete it from `sqlite_db.py`?). From looking at the copy_from post-hook, it looks like it will copy the table definitions there automatically - is that right? So if so, does the developer who changes a column name need to do anything with this copying, or is this all 'just for information'?

Could you provide some more details, or possibly discuss it with me on Slack sometime?

Reply:
I think my explanation is a bit confusing. What I meant is as follows: Post-hook copies the correct tables to the migration script, but it is not enough to have these classes. Let me explain it in a code snippet:

```python
class Platform(BaseSpatiaLite, PlatformMixin):
    identifier = deferred(Column(String(10), nullable=False))

def upgrade():
        with op.batch_alter_table("Platforms", schema=None, copy_from=Platform.__table__) as batch_op:
            batch_op.alter_column(column_name="pennant", new_column_name="identifier", nullable=False)

def downgrade():
        with op.batch_alter_table("Platforms", schema=None, copy_from=Platform.__table__) as batch_op:
            batch_op.alter_column(column_name="identifier", new_column_name="pennant", nullable=True)
```

As you can see that, in `upgrade()` function, Alembic tries to drop a column named "**pennant**". On the other hand, in `downgrade()` function, it tries to drop a column named "**identifier**". So, if you don't implement a different class for `upgrade()`, it is going to fail because it will not find a column named "**pennant**" (it takes it from `copy_from` parameter). The correct implementation should be like this:
```python
def upgrade():
        class Platform(BaseSpatiaLite, PlatformMixin):
            pennant = deferred(Column(String(10), nullable=False))
        with op.batch_alter_table("Platforms", schema=None, copy_from=Platform.__table__) as batch_op:
            batch_op.alter_column(column_name="pennant", new_column_name="identifier", nullable=False)

def downgrade():
        class Platform(BaseSpatiaLite, PlatformMixin):
            identifier = deferred(Column(String(10), nullable=False))
        with op.batch_alter_table("Platforms", schema=None, copy_from=Platform.__table__) as batch_op:
            batch_op.alter_column(column_name="identifier", new_column_name="pennant", nullable=True)
```

If it's still not clear, we can discuss it on Slack. I'll edit the documentation after we agreed on a good explanation.

Reply:
I'm still a bit confused about this: surely if we define a class inside the function with just one member variable, then it'll think the class only has one field? Or is this just a simple example with only one member variable?

We should probably discuss this on Slack once you're back.

Reply:
I'm sorry that I've seen this message now. It was just a simple example to show you the difference.